src/js/client/common-dialogs-widget.js

```diff
@@ -0,0 +1,124 @@
+
+/**
+ * DevPMT Base Dialog
+ *
+ * Creates the simplest dialog, currently based on the foundation
+ * modal dialog. Can be inherited to create more complex dialogs.
+ */
+fluid.defaults("gpii.devpmt.dialogs.baseDialog", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    templates: {
+        initial: "common-dialog-widget"
+    },
+    model: {},
+    invokers: {
+        openDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.openDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        },
+        closeDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.closeDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        }
+    },
+    selectors: {
+        dialogContainer: "#pmt-dialog-container"
+    },
+    listeners: {
+        "onMarkupRendered.openDialog": {
+            func: "{that}.openDialog"
+        }
+    },
+    events: {
+        // Sometimes foundation can move around the markup for the
+        // dialog to float it, etc. and we can lose event bindings.
+        // This allows activity to happen immediately after the dialog
+        // is opened.
+        afterOpenDialog: null
+    }
+});
+
+gpii.devpmt.dialogs.baseDialog.openDialog = function (that, dialogContainer) {
+    /* eslint-disable no-new */
+    // This foundation plugin requires using the `new` operator to work.
+    new Foundation.Reveal(dialogContainer, {
+        appendTo: that.options.selectors.initial
+    });
+    /* eslint-enable no-new */
+    dialogContainer.on("closed.zf.reveal", function () {
+        that.destroy();
+    });
+    dialogContainer.foundation("open");
+    that.events.afterOpenDialog.fire();
+};
+
+gpii.devpmt.dialogs.baseDialog.closeDialog = function (that, dialogContainer) {
+    dialogContainer.foundation("close");
+};
+
+/**
+ * DevPMT Confirmation Dialog
+ *
+ * Suitable dialog for performing actions that require either an
+ * 'accept' or 'cancel' type answer. Such as adding a product to
+ * an NP Set, deleting a context from an NP Set, etc.
+ *
+ * To create confirmation dialogs, inherit this grade. Then the
+ * following items need to be overriden:
+ *
+ * - Create a handlebars template in the standard partials area
+ * - Override the `templates.initial` option with that template.
+ * - The template need only retain the class names for the open
+ *   and close buttons in this grades selectors, to allow wiring.
+ * - You can optionally override the `acceptConfirmDialog` and
+ *   `cancelConfirmDialog` invokers with custom behavior, and
+ *   then close the dialog if your processing is successful.
+ */
+fluid.defaults("gpii.devpmt.dialogs.confirmDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.baseDialog"],
+    templates: {
+        initial: "common-confirm-dialog-widget"
+    },
+    selectors: {
+        acceptButton: ".pmt-confirm-dialog-button",
+        cancelButton: ".pmt-cancel-dialog-button"
+    },
+    invokers: {
+        acceptConfirmDialog: {
```

Comment:
You can get rid of definitions for functions `gpii.devpmt.dialogs.acceptConfirmDialog()` and `gpii.devpmt.dialogs.cancelConfirmDialog()` using:

```
invokers: {
    acceptConfirmDialog: "{that}.closeDialog",
    cancelConfirmDialog: "{that}.closeDialog"
    }
}
```
Also noticed all components that inherit "gpii.devpmt.dialogs.confirmDialog" have their own "acceptConfirmDialog" invoker implemented. If this function is meant to be implemented by integrators rather than performing a dialog close, this way can be considered:
```
invokers: {
    acceptConfirmDialog: "fluid.notImplemented",  // must be supplied by integrators
    cancelConfirmDialog: "{that}.closeDialog"
}
```
Please pick the one that makes sense to you. Thanks.

Reply:
this has been done